### PR TITLE
Delete code related to LoaderOptimization and SharedDomain

### PR DESCRIPTION
**Pull request by jkotas**
No longer needed

### CODE REVIEW

File: src/vm/assembly.cpp [hunk 1]
```diff
@@ -2349,49 +2249,6 @@ void DECLSPEC_NORETURN Assembly::ThrowBadImageException(LPCUTF8 pszNameSpace,
...
-    // TypeLibs are refcounted pointers.
-    if (pNew != pOld)
-    {
-        if (pNew && pNew != (ITypeLib*)-1)
-            pNew->AddRef();
-        if (pOld && pOld != (ITypeLib*)-1)
-            pOld->Release();
-    }
-} // void Assembly::SetTypeLib()
-
```
Comment (davidwrighton) [inline src/vm/assembly.cpp]: @AaronRobinsonMSFT  Do we need these for the COM interop support around IDispatch that you've been doing?

File: src/vm/assembly.cpp [hunk 2]
```diff
@@ -2349,49 +2249,6 @@ void DECLSPEC_NORETURN Assembly::ThrowBadImageException(LPCUTF8 pszNameSpace,
 
 
 #ifdef FEATURE_COMINTEROP
```
Comment (AaronRobinsonMSFT) [inline src/vm/assembly.cpp]: I assume we should also remove the `FEATURE_COMINTEROP` as well?

Comment (jkotas) [reply src/vm/assembly.cpp]: There is more COMINTEROP stuff under this ifdef.

File: src/vm/compile.cpp [hunk 3]
```diff
@@ -4702,38 +4666,16 @@ CORINFO_METHOD_HANDLE CEEPreloader::NextUncompiledMethod()
...
     if (m_uncompiledMethods.GetCount() == 0)
     {
-#ifdef FEATURE_FULL_NGEN
-        if (!m_fSpeculativeTriage)
         {
-            // We take one shot at smarter elimination of speculative instantiations
-            // that are guaranteed to be found in other modules
-            TriageSpeculativeInstantiations();
-            m_fSpeculativeTriage = TRUE;
+            // The subsequent populations are done in non-expansive way (won't load new types)
```
Comment (janvorli) [inline src/vm/compile.cpp]: A nit - the scope around this is not needed

File: src/vm/methoditer.cpp [hunk 4]
```diff
@@ -47,81 +47,48 @@ BOOL LoadedMethodDescIterator::Next(
...
-        if (m_assemblyIterationMode == kModeUnsharedADAssemblies)
-        {
-            // We're supposed to ignore shared assemblies, so check for them now
-            if ((*pDomainAssemblyHolder)->GetAssembly()->IsDomainNeutral())
-            {
-                goto ADVANCE_ASSEMBLY;
-            }
-        }
+    // We're supposed to ignore shared assemblies, so check for them now
+    if ((*pDomainAssemblyHolder)->GetAssembly()->IsDomainNeutral())
```
Comment (janvorli) [inline src/vm/methoditer.cpp]: It seems we can remove this as IsDomainNeutral always returns FALSE now, right?

**Comment by AaronRobinsonMSFT** [comment]
@dotnet-bot test OSX10.12 x64 Checked Innerloop Build and Test please

**Comment by jkotas** [comment]
I am not able to trace down what is causing the failure in "Ubuntu x64 Checked CoreFX Tests". I am going to submit this as multiple smaller PRs to get the bottom of it.

**Comment by jkotas** [comment]
@dotnet-bot test CentOS7.1 x64 Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test please
@dotnet-bot test Windows_NT x64 Release CoreFX Tests please

**Comment by jkotas** [comment]
@dotnet-bot test Ubuntu arm Cross Checked Innerloop Build and Test please
@dotnet-bot test Ubuntu arm Cross Checked no_tiered_compilation_innerloop Build and Test please

**Pull request merged**